app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -597,11 +594,8 @@ private void updateMapFragment(boolean isSlightUpdate) {
                         .observeOn(AndroidSchedulers.mainThread())
                         .subscribe(this::populatePlaces,
                                 throwable -> {
-                                    if (throwable instanceof UnknownHostException || throwable instanceof ConnectException) {
-                                        showErrorMessage(getString(R.string.slow_internet));
-                                    } else {
-                                        showErrorMessage(throwable.getMessage());
-                                    }
+                                    Timber.d(throwable);
+                                    showErrorMessage(getString(R.string.error_fetching_nearby_places));
```

Comment:
How about adding a `TODO: give meaningful error messages for each specific exception that might occur` here thus documenting the fact that this has to re-visited. It might even kindle people who accidentally see the TODO to work on it :wink: